src/setup_player.c

```diff
@@ -0,0 +1,39 @@
+#include <cub3D.h>
+
+static t_weapon	*setup_weapon(void);
+
+t_player	*setup_player(t_map *map)
+{
+	t_player	*player;
+
+	player = malloc(sizeof(t_player));
```

Comment:
what about unprotected mallocs, do we want to accept these?

Reply:
My Idea was to go through the Code at the end of production and add a malloc check + free after every malloc, so we don't have to constantly refactor all the dynamic frees